pkg/apis/clusterlink.net/v1alpha1/operator_types.go

```diff
@@ -0,0 +1,131 @@
+
+const (
+	// IngressTypeNone indicates that the deployment operator did not create any external ingress.
+	IngressTypeNone IngressType = "none"
+	// IngressTypeNodePort indicates that the deployment operator created an external ingress of type NodePort.
+	IngressTypeNodePort IngressType = "NodePort"
+	// IngressTypeLoadBalancer indicates that the deployment operator created an external ingress of type LoadBalancer.
+	IngressTypeLoadBalancer IngressType = "LoadBalancer"
+)
+
+// DataplaneType represents the dataplane type  of the deployed ClusterLink.
+type DataplaneType string
+
+const (
+	// GoType indicates that the dataplane type is Go-dataplane.
+	GoType DataplaneType = "go"
+	// EnvoyType indicates that the dataplane type is Envoy.
+	EnvoyType DataplaneType = "envoy"
+)
+
+// OperatorSpec defines the desired state of a ClusterLink Operator.
+type OperatorSpec struct {
+	DataPlane DataPlaneSpec `json:"dataplane,omitempty"`
+	Ingress   IngressSpec   `json:"ingress,omitempty"`
+
+	// +kubebuilder:validation:Enum=trace;debug;info;warning;error;fatal
+	// +kubebuilder:default=info
+	// LogLevel define the ClusterLink components log level.
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// +kubebuilder:default="ghcr.io/clusterlink-net"
+	// ContainerRegistry is the container registry to pull the ClusterLink project images.
+	ContainerRegistry string `json:"containerRegistry,omitempty"`
+	// +kubebuilder:default="latest"
+	// ImageTag represents the version of the ClusterLink project images.
+	ImageTag string `json:"imageTag,omitempty"`
+}
+
+// OperatorStatus defines the observed state of ClusterlLink Operator.
+type OperatorStatus struct {
+	Controlplane ComponentStatus `json:"controlplane,omitempty"`
+	Dataplane    ComponentStatus `json:"dataplane,omitempty"`
+}
+
+// ComponentStatus defines the status of component in ClusterLink.
+type ComponentStatus struct {
+	// +kubebuilder:validation:Enum=none;ready;error;pending
+	// +kubebuilder:default=none
+	Status  StatusType `json:"status,omitempty"`
+	Message string     `json:"message,omitempty"`
+}
+
+// DataPlaneSpec defines the desired state of the dataplane components in ClusterLink.
+type DataPlaneSpec struct {
+	// +kubebuilder:validation:Enum=envoy;go
+	// +kubebuilder:default=envoy
+	// Type represents the dataplane type. Supports values "go" and "envoy".
+	Type DataplaneType `json:"type,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=10
+	// +kubebuilder:default=1
+	// Replicas represents the number of dataplane replicas.
+	Replicas int `json:"replicas,omitempty"`
+}
+
+// IngressSpec defines the type of the ingress component in ClusterLink.
+type IngressSpec struct {
+	// +kubebuilder:validation:Enum=none;LoadBalancer;NodePort
+	// +kubebuilder:default=none
+	// Type represents the type of service used to expose the ClusterLink deployment. Supported values: "LoadBalancer","NodePort", "none".
+	Type IngressType `json:"type,omitempty"`
```

Comment:
what will be the name generated for the LoadBalancer of NodePort? User configured, or not? what are the defaults

Reply:
I updated the comment in the types